From da5ea799e1f22f16ecab5fc0f88d9607104abd3b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:13 -0400
Subject: [powerpc] powernv: Increase candidate fw image size

Message-id: <1410545655-205645-265-git-send-email-dzickus@redhat.com>
Patchwork-id: 94544
O-Subject: [RHEL7 PATCH 264/626] powerpc/powernv: Increase candidate fw image size
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit bf16a4c2515ace64f08f7adb182ca147c5b01145
Author: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
Date:   Thu Jan 2 17:00:42 2014 +0530

    powerpc/powernv: Increase candidate fw image size

    At present we assume candidate image is <= 256MB. But in P8,
    candidate image size can go up to 750MB. Hence increasing
    candidate image max size to 1GB.

    Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/opal-flash.c b/arch/powerpc/platforms/powernv/opal-flash.c
index 4aeae4f..16e571b 100644
--- a/arch/powerpc/platforms/powernv/opal-flash.c
+++ b/arch/powerpc/platforms/powernv/opal-flash.c
@@ -76,8 +76,8 @@
 /* Validate buffer size */
 #define VALIDATE_BUF_SIZE 4096
 
-/* XXX: Assume candidate image size is <= 256MB */
-#define MAX_IMAGE_SIZE 0x10000000
+/* XXX: Assume candidate image size is <= 1GB */
+#define MAX_IMAGE_SIZE 0x40000000
 
 /* Flash sg list version */
 #define SG_LIST_VERSION (1UL)
-- 
1.7.1